backend/sublet/views.py

```diff
@@ -292,15 +301,16 @@ def create(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
-    def destroy(self, request, *args, **kwargs):
+    def destroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         queryset = self.get_queryset()
         filter = {"user": self.request.user.id, "sublet": int(self.kwargs["sublet_id"])}
-        obj = get_object_or_404(queryset, **filter)
+        obj: Offer = get_object_or_404(queryset, **filter)
```

Comment:
What dictates if you explicitly annotate the type or not?

Reply:
If the type of a variable cant be easily inferred (like x=5 is obv an int) or is ambiguous (None or type can change), it might require an explicit annotation. Or some packages we use may not use type annotations, so when we call those functions, we may need to specify. Or, sometimes types may be ambiguous, like ints and floats are often convertible or strs and datetimes if in right format, so you might want to declare type to increase guardrailing